src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Acquisition/MissingSetupComponentRegistrationService.cs

```diff
@@ -308,17 +308,18 @@ private async Task DisplayMissingComponentsPromptAsync()
             return null;
         }
 
+        // Values in this dictionary must be List<string> within this method.
         Dictionary<Guid, IReadOnlyCollection<string>> vsComponentIdsToRegister = new();
```

Comment:
Any reason this can't be a `Dictionary<Guid, List<string>>` to begin with?

Reply:
Yeah, a limitation on the type system. `IMissingComponentRegistrationService.RegisterMissingComponentsAsync` accepts `IReadOnlyDictionary<Guid, IReadOnlyCollection<string>>`, and we can't cast `Dictionary<Guid, IReadOnlyCollection<string>>` to that (there's no variance on the `TValue` type argument here). I'm trying to avoid having a copy, so do the weird casting thing.